Gemfile

```diff
@@ -2,6 +2,7 @@ source "https://rubygems.org"
 
 gem "rails", "~> 5.2"
 
+gem "active_model_serializers", "~> 0.10.0"
```

Comment:
I wonder if we should be using this gem, given [its maintenance status](https://github.com/rails-api/active_model_serializers#status-of-ams). We do use it in Content Data API, but then we also use JBuilder in that app, as well as a few other apps. The other advantage of using JBuilder for me is that we can have a `show.json.jbuilder` in `app/views` i.e. we're not introducing another top-level `app/` folder.

Arguably we don't even need a gem for this one use case, as [it can just be done in a view](https://stackoverflow.com/questions/2088280/in-rails-how-do-you-render-json-using-a-view). Can we try that and see if it's a bit simpler?

Reply:
What are your specific concerns about the maintenance status?

Using ERB templates to handcraft JSON responses can easily become more complex and more error prone, especially if the response contains values that are collections, or dynamic in type (e.g. sometimes being nil).  I really don't want to introduce a pattern that's difficult to maintain particularly if the responses may change or become more complex.

Reply:
Perhaps I'm over-thinking here, and we can continue as-is. When I wrote my comments, I was thinking about my own approach when adding a new gem:

- I usually check the check the gem is under active maintenance; this one hasn't had any active development [for around 3 years](https://github.com/rails-api/active_model_serializers/commits/master). The risk there is that changes to other gems or Ruby itself could make it unusable and we'd have to switch to something else, or we'd have to fork it in order to get it back into a working state.

- If a gem is looking under-maintained, I'd then check the issues and README, as sometimes code just doesn't need changing. This one currently has 141 open issues (not high, but not great), and a "ActiveModelSerializers is undergoing some renovations." line that suggests things will break for us in the future.

My conclusion was that I wouldn't add this gem to a repo, and would look for an alternative. The alternative ERB suggestion was a bit far out, though - I hadn't thought that through, sorry! I do recommend JBuilder, as it fares much better with the above heuristics.

Since we already use both of these on GOV.UK, we've kind of committed to them anyway, so there's a reasonable argument that we're just repeating an existing pattern.

Reply:
Yup - totally understand the concern there, I think its a little confusing. The gem version we're using is based off the `0-10-stable` branch, which appears to be maintained (200~ commit ahead of master). Also I feel that gem has somewhat been tried and tested with 40K~ projects having used it. No worries!